authorization-server/utils/redis.js

```diff
@@ -1,12 +1,34 @@
 import { createClient } from 'redis';
 
-const createRedisClient = async () => {
-  const client = createClient({
-    url: process.env.REDIS_SERVER,
-  });
-  client.on('error', (err) => console.log('Redis Client Error', err));
+const redisClient = createClient({
```

Comment:
this is going to cause you problems.  Performing this sort of logic at the top level is an anti pattern.  Connecting and construction of the RedisConnection object below is a responsibility of the server.